src/compiler/program.ts

```diff
@@ -1551,6 +1566,30 @@ namespace ts {
             }
         }
 
+        function processLibReferenceDirectives(file: SourceFile) {
+            const libDirectory = getLibDirectory();
+            forEach(file.libReferenceDirectives, libReference => {
+                const libName = libReference.fileName.toLocaleLowerCase();

```

Comment:
Do we really need to lower-case it? If they try accessing `/// <reference lib="es2017.Object" />` we will correct their spelling to `"es2017.object"`.